Framework/HistogramData/inc/MantidHistogramData/Histogram.h

```diff
@@ -192,8 +192,12 @@ class MANTID_HISTOGRAMDATA_DLL Histogram {
   void convertToCounts();
   void convertToFrequencies();
 
-  HistogramIterator begin() const;
-  HistogramIterator end() const;
+  HistogramIterator begin() const &;
+  HistogramIterator end() const &;
+  // Calling begin/end on a temporary is not allowed as the iterator
+  // reference will be immediately invalidated.
+  HistogramIterator begin() const && = delete;
+  HistogramIterator end() const && = delete;
```

Comment:
I don't think you actually need to explicitly do that, havinh the `&` should be sufficient. Just like you do not need to delete the non-const overload of a method if there is only a const version?

Reply:
I tried that first, put back the original code in `LineProfile` and everything still compiled, which surprised me too. I will double check.